### PR TITLE
Integrating Terraform into salvo-remote.

### PR DESCRIPTION
**Pull request by mum4k**
The `salvo-remote` binary is now able to deploy a Sandbox using Terraform. Done here:

- added required Go repositories into the `WORKSPACE`.
- renamed `BUILD` to `BUILD.bazel` which is what `gazelle` uses by default.
- moved `bazel_gazelle` dependency up to the `WORKSPACE` file as it is needed in multiple separate `.bzl` files.
- added a patch that manually enables `cgo` for `github.com/cloudflare/circl` as per https://github.com/bazelbuild/bazel-gazelle/issues/1421.
- added a new package `sandboxes` that manages sandbox instances.
- validated `salvo-remote` flags and deploying a sandbox using the `sandboxes` package.

### CODE REVIEW

File: salvo-remote/salvo_remote.go [hunk 1]
```diff
@@ -3,16 +3,68 @@
...
+	return nil
+}
+
+// mainWithErrCode returns an error code after execution.
+func mainWithErrCode() int {
 	flag.Parse()
-	fmt.Printf("salvo-remote not implemented yet, executed with -build_id %q and -build_id_override %q\n", *buildID, *buildIDOverride)
-	os.Exit(1)
+	if err := runSalvoRemote(sandboxes.NewManager()); err != nil {
+		fmt.Printf("salvo-remote: %v\n", err)
```
Comment (gyohuangxin) [inline salvo-remote/salvo_remote.go:62]: ditto

File: salvo-remote/salvo_remote.go [hunk 2]
```diff
@@ -3,16 +3,68 @@
...
+
+// sandboxManager abstracts a type that manages sandbox instances.
+type sandboxManager interface {
+	Start(context.Context, map[sandboxes.Type]sandboxes.Instances) error
+}
+
+// runSalvoRemote executes Salvo remotely and returns the exit code.
+func runSalvoRemote(sm sandboxManager) error {
+	if err := validateFlags(); err != nil {
+		return fmt.Errorf("validateFlags => %v", err)
```
[the remark this answers is not in the document]
Comment (mum4k) [reply salvo-remote/salvo_remote.go:44]: The `%v` in the text actually contains the human readable error message.

When printed out on the screen, the full text will look like this:

```
validateFlags => got -build_id(-1), -build_id must be set to a positive integer
```

File: salvo-remote/salvo_remote.go [hunk 3]
```diff
@@ -3,16 +3,68 @@
...
+	}
+
+	sbxs := map[sandboxes.Type]sandboxes.Instances{
+		sandboxes.TypeDefaultSandboxX64: {getBuildID()},
+	}
+	ctx := context.Background()
+
+	if err := sm.Start(ctx, sbxs); err != nil {
+		return fmt.Errorf("sm.Start => %v", err)
```
[the remark this answers is not in the document]
Comment (mum4k) [reply salvo-remote/salvo_remote.go:53]: Please see my response above.

File: salvo-remote/salvo_remote.go (hunk 1 above)
Comment (mum4k) [reply salvo-remote/salvo_remote.go:62]: Please see my response above.

File: salvo-remote/salvo_remote.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (gyohuangxin) [reply salvo-remote/salvo_remote.go:44]: Understand.

**Review by gyohuangxin — APPROVED** [review]
LGTM, thanks!

File: salvo-remote/sandboxes/sandboxes.go [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+	return "TypeUnknown"
+}
+
+// typeNames maps Type values to human readable names.
+var typeNames = map[Type]string{
+	TypeDefaultSandboxX64: "default_sandbox_x64_build_ids",
+}
+
+const (
+	typeUnknown Type = iota
```
Comment (fei-deng) [inline salvo-remote/sandboxes/sandboxes.go:37]: Just curious, is `typeUnknown` being used anywhere?

**Review by fei-deng — APPROVED** [review]

File: salvo-remote/sandboxes/sandboxes.go (hunk 4 above)
Comment (mum4k) [reply salvo-remote/sandboxes/sandboxes.go:37]: `typeUnknown` would get used if someone passed in the default value of the type `sandboxes.Type`.

Defining it here helps in isolating that sort of programming mistake, same as when we have an enum in a proto file and we assign the first value to UNKNOWN.

**Comment by armfazh** [comment]
The compilation issues of CIRCL are solved by updating [rules_go v0.55.0](https://github.com/bazel-contrib/rules_go/releases/tag/v0.55.0). See more details at CIRCL [Bazel-Compilation](https://github.com/cloudflare/circl/wiki/Bazel-Compilation) wiki.

**Pull request merged**